superset/viz.py

```diff
@@ -554,46 +555,50 @@ def query_obj(self):
 
         # Add all percent metrics that are not already in the list
         if "percent_metrics" in fd:
-            d["metrics"] = d["metrics"] + list(
-                filter(lambda m: m not in d["metrics"], fd["percent_metrics"] or [])
+            d["metrics"].extend(
+                m for m in fd["percent_metrics"] or [] if m not in d["metrics"]
```

Comment:
no action required: I'm surprised to not see brackets around the comprehension, guessing it just returns a generator at that point and `.extend` is ok with that.

Reply:
@mistercrunch [`list.extend(iter)`](https://docs.python.org/3/tutorial/datastructures.html#more-on-lists) takes any iterable.